rmgpy/rmg/main.py

```diff
@@ -1463,21 +1481,126 @@ def generateCanteraFiles(self, chemkinFile, **kwargs):
-            # No need to initialize reaction threshold arrays in this case
+                self.trimolecularThreshold = np.zeros((numCoreSpecies, numCoreSpecies, numCoreSpecies),bool)
+
+            if self.restart:
+                # Load in the restart mapping
+                with open(os.path.join(self.speciesMapPath), 'r') as f:
+                    restartSpeciesMap = yaml.safe_load(stream=f)
+
+                numRestartSpcs = len(restartSpeciesMap)
+
+                for index, adj_list in restartSpeciesMap.items():
+                    restartSpeciesMap[index] = Species().fromAdjacencyList(adj_list)
+
+                # Load in the restart filter tensors
+                with h5py.File(self.filtersPath, 'r') as f:
+                    try:
+                        unimolecularThreshold_restart = f.get('unimolecularThreshold').value
+                        bimolecularThreshold_restart = f.get('bimolecularThreshold').value
+                        if self.trimolecular:
+                            trimolecularThreshold_restart = f.get('trimolecularThreshold').value
+
+                        # Expand Thresholds to match number of species in the current model.
+                        unimolecularThreshold = np.zeros(numCoreSpecies, bool)
+                        unimolecularThreshold[:numRestartSpcs] = unimolecularThreshold_restart
+
+                        bimolecularThreshold = np.zeros((numCoreSpecies, numCoreSpecies), bool)
+                        bimolecularThreshold[:numRestartSpcs, :numRestartSpcs] = bimolecularThreshold_restart
+
+                        if self.trimolecular:
+                            trimolecularThreshold = np.zeros((numCoreSpecies, numCoreSpecies, numCoreSpecies), bool)
+                            trimolecularThreshold[:numRestartSpcs, :numRestartSpcs, :numRestartSpcs] = \
+                                trimolecularThreshold_restart
+
+                        filters_found = True
+
+                    except KeyError:  # If we can't find the filters then this is because filtering was not used
+                        logging.warning('No filters were found in file {0}. This is to be expected if the restart '
+                                        'files specified came from an RMG job that did not use filtering. Therefore, '
+                                        'RMG will assume that all of the core species from the restart seed have '
+                                        'already been reacted and will not react them again. Additional species that '
+                                        'were added to the input file but are not in the restart core seed WILL be '
+                                        'reacted with the rest of the core though. If this is not the intended '
+                                        'behavior please stop this job and correct the missing '
+                                        'filters.'.format(self.filtersPath))
```

Comment:
`came from an RMG job that did not use filtering` -> `are from an RMG job without reaction filtering`
`species that were added to the input file but are not` -> `species added to the input file but not`
`core though.` -> `core.`
`restart seed` vs `restart core seed` should probably be consistent?
I think the last sentence is probably not needed. I don't know how many people read the log in real time.

Reply:
Thanks!